### PR TITLE
[MRG] MAINT: p_secs -> sections

### LINKED ISSUE #417: rename p_syn and p_secs
This is more a note to myself to remember to rename `p_syn` and `p_secs` in the `Cell` object to `synapses` and `sections`. And potentially reconsider creating objects for them, but at the very least rename the neuron objects to be `_nrn_sections` etc.

### PR DESCRIPTION
**Pull request by jasmainak**
closes #417

### CODE REVIEW

File: hnn_core/cell.py [hunk 1]
```diff
@@ -232,15 +227,16 @@ class Cell:
...
-        self.p_secs = p_secs
-        self.p_syn = p_syn
+        self.sections = sections
+        self.synapses = synapses
         self.topology = topology
         self.sect_loc = sect_loc
-        self.sections = dict()
-        self.synapses = dict()
+        self._nrn_sections = dict()
+        self._nrn_synapses = dict()
```
Comment (rythorpe) [inline hnn_core/cell.py:239]: Good call. This helps us more readily know exactly where Neuron objects are stored.

File: hnn_core/cell.py [hunk 2]
```diff
@@ -189,11 +189,6 @@ class Cell:
     ----------
     pos : list of length 3
         The position of the cell.
-    sections : dict
-        The sections. The key is the name of the section
-        and the value is an instance of h.Section.
-    synapses : dict
-        The synapses that the cell can use for connections.
```
Comment (rythorpe) [inline hnn_core/cell.py:196]: Why were these removed from the docstring? They're still attributes right?

Comment (jasmainak) [reply hnn_core/cell.py]: Good catch! Fixed now

**Review by rythorpe — APPROVED** [review]

**Comment by rythorpe** [comment]
Thanks @jasmainak!
